Fix stale info about spread minDomains (#42758)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md

  nodes match the node selector.

  {{< note >}}

>   The `MinDomainsInPodTopologySpread` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enables `minDomains` in pod 
>   topology spread. From v1.28 release, `MinDomainsInPodTopologySpread` gate 

Reviewer: ```suggestion
  The `MinDomainsInPodTopologySpread` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
  enables `minDomains` for pod topology spread. Starting from v1.28,
  the `MinDomainsInPodTopologySpread` gate 
```

  nodes match the node selector.

  {{< note >}}
  The `MinDomainsInPodTopologySpread` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enables `minDomains` in pod 
  topology spread. From v1.28 release, `MinDomainsInPodTopologySpread` gate 

>   is enabled by default and older clusters it might have explicitly 

Reviewer: This sentence doesn't flow well, pls double check.
